File: ui/src/lib/model/GradeDistribution.ts
import type {Course} from "$lib/model/Instructor";

export type GradeDistribution = {
    course: Course;
    term: string;
    year: number;
    classAverage: number;
    classSize: number;
    grades: Grade[];
};

type Grade = {
    grade: string;
    count: number;
};

File: backend/src/main/java/courses/concordia/model/GradeDistribution.java
package courses.concordia.model;

import lombok.*;
import lombok.experimental.Accessors;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@Accessors(chain = true)
@Document(collection = "gradeDistributions")
public class GradeDistribution {
    @MongoId
    private String _id;
    private Course course;
    private String term;
    private int year;
    private int classAverage;
    private int classSize;
    private List<Grade> grades;

    @Data
    @AllArgsConstructor
    public static class Grade {
        private String grade;
        private int count;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Course {
        private String subject;
        private String catalog;
    }
}